Interesting stuff. Thanks!

I must admit that @ SURF we switched to opentelemetry for tracing; I'm willing to merge + release this.

Only some flake8 stuff:
```
flake8 run-test: commands[0] | flake8 starlette_opentracing tests
starlette_opentracing/middleware.py:23:43: W504 line break after binary operator
starlette_opentracing/middleware.py:24:41: W504 line break after binary operator
```

--- a/starlette_opentracing/middleware.py
+++ b/starlette_opentracing/middleware.py
@@ -17,13 +17,15 @@ class StarletteTracingMiddleWare(BaseHTTPMiddleware):
     def get_template(self, request: Request) -> str:
         """Get the template for the route endpoint."""
         method = request.method
-        urls = [
-            route
-            for route in request.scope["router"].routes
-            if hasattr(route, "endpoint") and
-            "endpoint" in request.scope and
-            route.endpoint == request.scope["endpoint"]
-        ]
+        if "endpoint" in request.scope:
+            urls = [
+                route
+                for route in request.scope["router"].routes
+                if hasattr(route, "endpoint") and route.endpoint == request.scope["endpoint"]
+            ]
+        else:
+            urls = []
+
         template = urls[0].path if len(urls) > 0 else "UNKNOWN"
         method_path = method + " " + template
         return method_path

--- a/tests/test_tracer.py
+++ b/tests/test_tracer.py
@@ -73,3 +73,21 @@ def test_tracer_uses_path_templates_for_operation_names():
     spans = mocked_tracer.finished_spans()
     assert len(spans) == 1
     assert spans[0].operation_name == "GET /foo/{foo_id}"
+
+
+def test_tracer_uses_set_template_to_unknown_if_no_endpoint():
+    app = Starlette()
+    mocked_tracer = MockTracer(scope_manager=ContextVarsScopeManager())
+    app.add_middleware(
+        StarletteTracingMiddleWare, tracer=mocked_tracer, use_template=True
+    )
+
+    @app.route("/foo/{foo_id}")
+    def foo(foo_id: str):
+        return PlainTextResponse(f"Foo: {foo_id}")
+
+    client = TestClient(app)
+    client.get("/bar/MyFoo")
+    spans = mocked_tracer.finished_spans()
+    assert len(spans) == 1
+    assert spans[0].operation_name == "GET UNKNOWN"

